### PR TITLE
Implement show taskbar icon option & appmenu on Wayland

### PR DESCRIPTION
**Pull request by ilya-fedin**

### CODE REVIEW

File: Telegram/SourceFiles/platform/linux/linux_wayland_integration.cpp [hunk 1]
```diff
@@ -92,10 +103,42 @@ WaylandIntegration::Private::Private()
...
+		});
+
+	connect(
+		&_applicationRegistry,
+		&Registry::plasmaShellAnnounced,
+		[=](uint name, uint version) {
+			_plasmaShell = {
+				_applicationRegistry.createPlasmaShell(name, version),
+				std::default_delete<PlasmaShell>(),
+			};
```
Comment (23rd) [inline Telegram/SourceFiles/platform/linux/linux_wayland_integration.cpp:119]: Why not use `std::make_unique`?

File: Telegram/SourceFiles/platform/linux/main_window_linux.cpp [hunk 2]
```diff
@@ -1266,9 +1281,11 @@ void MainWindow::handleVisibleChangedHook(bool visible) {
 #ifndef DESKTOP_APP_DISABLE_DBUS_INTEGRATION
 	if (_appMenuSupported && _mainMenuExporter) {
 		if (visible) {
-			RegisterAppMenu(winId(), kMainMenuObjectPath.utf16());
+			base::call_delayed(1, this, [=] {
+				RegisterAppMenu(windowHandle(), kMainMenuObjectPath.utf16());
+			});
```
Comment (23rd) [inline Telegram/SourceFiles/platform/linux/main_window_linux.cpp:1286]: Why this hack is necessary?

Comment (ilya-fedin) [reply Telegram/SourceFiles/platform/linux/main_window_linux.cpp:1286]: handleVisibleChanged seem to be called before native surface is created

File: Telegram/SourceFiles/platform/linux/linux_wayland_integration.h [hunk 3]
```diff
@@ -16,6 +16,12 @@ class WaylandIntegration {
 	void waitForInterfaceAnnounce();
 	bool supportsXdgDecoration();
 	QString nativeHandle(QWindow *window);
+	bool skipTaskbarSupported();
```
Comment (23rd) [inline Telegram/SourceFiles/platform/linux/linux_wayland_integration.h:19]: If you use the `[[nodiscard]]` attribute in the `WaylandIntegration::Private` class, then why the same attribute is skipped here in the `WaylandIntegration` class?

File: Telegram/SourceFiles/platform/linux/linux_wayland_integration.cpp (hunk 1 above)
Comment (ilya-fedin) [reply Telegram/SourceFiles/platform/linux/linux_wayland_integration.cpp:119]: How can I use it here?

File: Telegram/SourceFiles/platform/linux/main_window_linux.cpp (hunk 2 above)
Comment (23rd) [reply Telegram/SourceFiles/platform/linux/main_window_linux.cpp:1286]: Can it be replaced with `InvokeQueued` from `lib_base/base/invoke_queued.h`?

Comment (ilya-fedin) [reply Telegram/SourceFiles/platform/linux/main_window_linux.cpp:1286]: Seem to work fine

File: Telegram/SourceFiles/platform/linux/main_window_linux.cpp [hunk 4]
```diff
@@ -1255,7 +1270,7 @@ void MainWindow::updateGlobalMenuHook() {
 
 void MainWindow::handleVisibleChangedHook(bool visible) {
 	if (visible) {
-		base::call_delayed(1, this, [=] {
+		InvokeQueued(this, [=] {
```
Comment (23rd) [inline Telegram/SourceFiles/platform/linux/main_window_linux.cpp:1273]: Does it work well?

Comment (ilya-fedin) [reply Telegram/SourceFiles/platform/linux/main_window_linux.cpp:1273]: It works at each hide/show on my machine, at least

File: Telegram/SourceFiles/platform/linux/linux_wayland_integration.cpp (hunk 1 above)
Comment (23rd) [reply Telegram/SourceFiles/platform/linux/linux_wayland_integration.cpp:119]: Hmm, looks like you can't.
But I think `std::default_delete<PlasmaShell>()` is redundant.

File: Telegram/SourceFiles/platform/linux/main_window_linux.cpp (hunk 4 above)
Comment (23rd) [reply Telegram/SourceFiles/platform/linux/main_window_linux.cpp:1273]: Good, thanks.

File: Telegram/SourceFiles/platform/linux/linux_wayland_integration.cpp (hunk 1 above)
Comment (ilya-fedin) [reply Telegram/SourceFiles/platform/linux/linux_wayland_integration.cpp:119]: It doesn't build with only one argument.

Comment (23rd) [reply Telegram/SourceFiles/platform/linux/linux_wayland_integration.cpp:119]: ?
```suggestion
			_plasmaShell = std::unique_ptr<PlasmaShell>(
				_applicationRegistry.createPlasmaShell(name, version));
```

**Pull request merged**